pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerImpl.java

```diff
@@ -952,6 +952,7 @@ void recycle() {
             rePopulate = null;
             sequenceId = -1;
             createdAt = -1;
+            numMessagesInBatch = 1;
```

Comment:
Rather than setting this here, we should set it correctly in the two `create()` methods

Reply:
Got it,I will remove it.